File: app/src/main/java/com/tripsta/tripstapoker/card/CardListAdapter.java
package com.tripsta.tripstapoker.card;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.tripsta.models.Card;
import com.tripsta.tripstapoker.R;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by georgiossarris on 15/07/16.
 */

public class CardListAdapter extends RecyclerView.Adapter<CardListAdapter.RecyclerViewHolders> {

	private List<Card> cards;
	private Context context;

	public CardListAdapter(Context context, List<Card> cards) {
		this.cards = cards;
		this.context = context;
	}

	@Override
	public CardListAdapter.RecyclerViewHolders onCreateViewHolder(ViewGroup parent, int viewType) {

		View layoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_view_list, null);
		RecyclerViewHolders rcv = new RecyclerViewHolders(layoutView);
		return rcv;
	}

	@Override
	public void onBindViewHolder(RecyclerViewHolders holder, int position) {
		int number = cards.get(position).getNumber();
		String stringedNumber = String.valueOf(number);
		if (number == -1) {
			stringedNumber = "1/2";
		}
		holder.textView.setTag(number);
		holder.textView.setText(stringedNumber);
		holder.textView.setBackgroundColor(ContextCompat.getColor(context, cards.get(position).getColorHex()));
	}

	@Override
	public int getItemCount() {
		return this.cards.size();
	}

	class RecyclerViewHolders extends RecyclerView.ViewHolder {

		@BindView(R.id.textView)
		TextView textView;

		RecyclerViewHolders(View itemView) {
			super(itemView);
			ButterKnife.bind(this, itemView);
		}
	}

}

File: app/src/main/java/com/sarris/poker/card/CardListAdapter.kt
package com.sarris.poker.card

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.sarris.models.Card
import com.sarris.poker.R
import com.sarris.poker.card.CardListAdapter.RecyclerViewHolders

/**
 * Created by georgiossarris on 15/07/16.
 */
class CardListAdapter(private val context: Context, cards: List<com.sarris.models.Card>) :
    RecyclerView.Adapter<RecyclerViewHolders>() {
    private val cards: List<Card> = cards

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolders {
        val layoutView = LayoutInflater.from(parent.context).inflate(R.layout.card_view_list, null)
        val rcv = RecyclerViewHolders(layoutView)
        return rcv
    }

    override fun onBindViewHolder(holder: RecyclerViewHolders, position: Int) {
        val number: Int = cards[position].number
        var stringedNumber = number.toString()
        if (number == -1) {
            stringedNumber = "1/2"
        }
        holder.textView!!.tag = number
        holder.textView!!.text = stringedNumber
        holder.textView!!.setBackgroundColor(
            ContextCompat.getColor(
                context,
                cards[position].colorHex
            )
        )
    }

    override fun getItemCount(): Int {
        return cards.size
    }

    inner class RecyclerViewHolders(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @JvmField
        @BindView(R.id.textView)
        var textView: TextView? = null

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
